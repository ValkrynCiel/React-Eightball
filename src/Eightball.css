.Eightball {
    width: 22em;
    height: 22em;
    border-radius: 11em;
    position: relative;
    border: 2px solid purple;
    margin-left: calc(50vw - 11em);
    margin-top: calc(50vh - 11em); 
}

.Eightball-text {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    font-size: 40px;
}