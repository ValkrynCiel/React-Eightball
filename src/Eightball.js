import React, { Component } from 'react';
import './Eightball.css';
import answers from './answers';

class Eightball extends Component {

    static defaultProps = { answers };

    constructor(props){
        super(props);
        this.state = { statement: "Think of a question",
                       color: "black" };
        
        this.handleClick = this.handleClick.bind(this);
    }

    getRandom() {
        return Math.floor(Math.random() * this.props.answers.length)
    }

    handleClick(evt) {
        let index = this.getRandom();
        this.setState({ statement: answers[index].msg,
                        color: answers[index].color,
                        });
    }

    render () {
        const colors = {backgroundColor: this.state.color,
                        color: "white"}
        return (
            <div className = "Eightball" 
                 onClick = { this.handleClick } 
                 style = {colors}>
                 <b className = "Eightball-text">{ this.state.statement } </b>
            </div>
        );
    }
}


export default Eightball;