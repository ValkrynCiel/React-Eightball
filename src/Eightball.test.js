import React from 'react';
import Eightball from './Eightball';
import toJson from "enzyme-to-json";
import { shallow, mount } from "enzyme";


it("renders using shallow", function() {
    shallow(<Eightball />);
  });
  
  it("renders using mount", function(){
    mount(<Eightball />);
  });
  
  it("matches snapshot", function() {
    let wrapper = shallow(<Eightball />);
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });


  it('can use output tests', function () {
    let wrapper = mount(<Eightball />);
  

  
    expect(wrapper.html()).toContain("Think of a question");
  });


  // { msg: "As I see it, yes.", color: "green" }
  it('can be props/state tested', function () {
    let wrapper = mount(<Eightball answers={[ { msg: "As I see it, yes.", color: "green" }]}/>);
  
    expect(wrapper.prop('answers')).toEqual([ { msg: "As I see it, yes.", color: "green" }]);
  
    wrapper.setProps({answers: [ { msg: "As I see it, yes.", color: "green" }]});
    console.log(wrapper.debug());
    expect(wrapper.html()).toContain("Think of a question");
  
    wrapper.setState({ statement: "As I see it, yes.", color: "green" });
    let b_elem = wrapper.find('b').first();
    // note {42}, not just 42 --- has to match exactly
    console.log('b element is ', b_elem);
    //expect(b_elem.equals(<b className = "Eightball-text">As I see it, yes.</b>)).toEqual(true);
    expect(wrapper.matchesElement(<b>As I see it, yes.</b>));
  });