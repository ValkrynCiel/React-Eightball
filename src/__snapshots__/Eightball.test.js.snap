// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="Eightball"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "black",
      "color": "white",
    }
  }
>
  <b
    className="Eightball-text"
  >
    Think of a question
     
  </b>
</div>
`;
